### PR TITLE
Add timesync drift testcase

### PR DESCRIPTION
**Pull request by dsrivastavv**
Add timesync drift testcase

### CODE REVIEW

File: microsoft/testsuites/core/timesync.py [hunk 1]
```diff
@@ -377,3 +380,55 @@ def verify_pmu_disabled_for_arm64(self, node: Node) -> None:
...
+                    and current_time.day == node_time.day
+                ):
+                    is_drift_corrected = True
+                    log.debug(
+                        f"Iteration {iteration}: Time drift is not corrected by "
+                        "chrony yet. Retrying..."
+                    )
+                    break
+
+            if not is_drift_corrected:
```
Comment (squirrelsc) [inline microsoft/testsuites/core/timesync.py]: Use `assert_that`

File: microsoft/testsuites/core/timesync.py [hunk 2]
```diff
@@ -377,3 +380,55 @@ def verify_pmu_disabled_for_arm64(self, node: Node) -> None:
...
+            # Get current time
+            date = node.tools[Date]
+            node_time = date.current()
+            modified_time = deepcopy(node_time)
+            modified_time = modified_time.replace(year=node_time.year - 2)
+            date.set(modified_time)
+
+            # Poll every second and check if the time drift is corrected
+            # for 5 minutes
+            for _ in range(300):
```
Comment (squirrelsc) [inline microsoft/testsuites/core/timesync.py]: Can this timeout to be shorter by some settings?

File: microsoft/testsuites/core/timesync.py [hunk 3]
```diff
@@ -377,3 +380,55 @@ def verify_pmu_disabled_for_arm64(self, node: Node) -> None:
...
+                    current_time.year == node_time.year
+                    and current_time.month == node_time.month
+                    and current_time.day == node_time.day
+                ):
+                    is_drift_corrected = True
+                    log.debug(
+                        f"Iteration {iteration}: Time drift is not corrected by "
+                        "chrony yet. Retrying..."
+                    )
+                    break
```
Comment (squirrelsc) [inline microsoft/testsuites/core/timesync.py:420]: The logic is weird. The debug message says it's not correct, but it breaks here.

File: microsoft/testsuites/core/timesync.py [hunk 4]
```diff
@@ -377,3 +380,55 @@ def verify_pmu_disabled_for_arm64(self, node: Node) -> None:
...
+            modified_time = modified_time.replace(year=node_time.year - 2)
+            date.set(modified_time)
+
+            # Poll every second and check if the time drift is corrected
+            # for 5 minutes
+            for _ in range(300):
+                time.sleep(1)
+                current_time = date.current()
+                if (
+                    current_time.year == node_time.year
```
Comment (squirrelsc) [inline microsoft/testsuites/core/timesync.py]: There maybe a bug. if the test case running at last minute a day, after 5 minutes, the day maybe not the same anyway. The year and month have the same problem. Please find another stable way to check it.

File: microsoft/testsuites/core/timesync.py (hunk 2 above)
Comment (dsrivastavv) [reply microsoft/testsuites/core/timesync.py]: Sure! The max I observed was around 60 sec for 3 iteration, so we can keep 120 seconds with some buffer and increase if we observe any errors

File: microsoft/testsuites/core/timesync.py (hunk 3 above)
Comment (dsrivastavv) [reply microsoft/testsuites/core/timesync.py:420]: Yeah. I mistakenly put the debug log in the if condition. Should be fixed now

File: microsoft/testsuites/core/timesync.py (hunk 4 above)
Comment (dsrivastavv) [reply microsoft/testsuites/core/timesync.py]: Fixed. Comparing datetime object directly now

File: microsoft/testsuites/core/timesync.py [hunk 5]
```diff
@@ -377,3 +380,47 @@ def verify_pmu_disabled_for_arm64(self, node: Node) -> None:
...
+
+            # Poll every second and check if the time drift is corrected
+            # for 5 minutes
+            for _ in range(120):
+                time.sleep(1)
+                current_time = date.current()
+                if current_time > node_time:
+                    is_drift_corrected = True
+                    break
+                log.debug(
```
Comment (squirrelsc) [inline microsoft/testsuites/core/timesync.py]: It outputs too many lines, output it every 10 seconds, or only once.

**Comment by squirrelsc** [comment]
@LiliDeng LGTM

**Review by LiliDeng — APPROVED** [review]

**Pull request merged**